tests/functionnal/Ticket.php

```diff
@@ -2912,4 +2916,79 @@ public function testMerge() {
       $this->integer((int)$supplier_count)->isEqualTo(3);
    }
+
+   public function testKeepScreenshotsOnFormReload() {
+      //login to get session
+      $auth = new \Auth();
+      $this->boolean($auth->login(TU_USER, TU_PASS, true))->isTrue();
+
+      // Test display of saved inputs from a previous submit
+      $_SESSION['saveInput'][\Ticket::class] = [
+         'content' => '&lt;p&gt; &lt;/p&gt;&lt;p&gt;&lt;img id="3e29dffe-0237ea21-5e5e7034b1d1a1.77230247"'
+         . ' src="data:image/png;base64,' . $this->inlineImage . '" width="12" height="12" /&gt;&lt;/p&gt;',
+      ];
+
+      $this->output(
+         function () {
+            $instance = new \Ticket();
+            $instance->showForm('-1');
+         }
+      )->contains('src="data:image/png;base64,' . $this->inlineImage . '"');
+   }
+
+   public function testScreenshotConvertedIntoDocument() {
+      $filename = '5e5e92ffd9bd91.67052325image_paste5541550.png';
+      $instance = new \Ticket();
+      $input = [
+         'name'    => 'a ticket',
+         'content' => '&lt;p&gt; &lt;/p&gt;&lt;p&gt;&lt;img id="3e29dffe-0237ea21-5e5e7034b1d1a1.77230247"'
+         . ' src="data:image/png;base64,' . $this->inlineImage . '" width="12" height="12" /&gt;&lt;/p&gt;',
+         '_content' => [
+            $filename,
+         ],
+         '_tag_content' => [
+            '3e29dffe-0237ea21-5e5e7034b1d1a1.77230247',
+         ],
+         '_prefix_content' => [
+            '5e5e92ffd9bd91.67052325',
+         ]
+      ];
+
+      $tmpFilename = GLPI_TMP_DIR . '/' . $filename;
+      file_put_contents($tmpFilename, base64_decode($this->inlineImage));
+
+      $instance->add($input);
+
+      // Ensure there is an anchor to the uploaded document
+      $expected = 'a href=&quot;/front/document.send.php?docid=';
+      $this->string($instance->fields['content'])->contains($expected);
+   }
+
+   public function testKeepScreenshotFromTemplate() {
+      //login to get session
+      $auth = new \Auth();
+      $this->boolean($auth->login(TU_USER, TU_PASS, true))->isTrue();
+
+      // create a template with a predeined description
+      $ticketTemplate = new \TicketTemplate();
+      $ticketTemplate->add([
+         'name' => $this->getUniqueString(),
+      ]);
+      $content = '&lt;p&gt;&lt;img id="3e29dffe-0237ea21-5e57d2c8895d55.57735524"'
+      . ' src="data:image/png;base64,' . $this->inlineImage . '" width="12" height="12" /&gt;&lt;/p&gt;';
+      $predefinedField = new \TicketTemplatePredefinedField();
+      $predefinedField->add([
+         'tickettemplates_id' => $ticketTemplate->getID(),
+         'num' => '21',
+         'value' => $content
+      ]);
+      $_SESSION['glpiactiveprofile']['tickettemplates_id'] = $ticketTemplate->getID();
+
+      $this->output(
+         function () {
+            $instance = new \Ticket();
+            $instance->showForm('0');
+         }
+      )->contains('src="data:image/png;base64,' . $this->inlineImage . '"');
```

Comment:
It is preferable to restore previous state of session before doing a assertions.
```suggestion
      $session_tpl_id_back = $_SESSION['glpiactiveprofile']['tickettemplates_id'];
      $_SESSION['glpiactiveprofile']['tickettemplates_id'] = $ticketTemplate->getID();

      $this->output(
         function () {
            $instance = new \Ticket();
            $instance->showForm('0');
            $_SESSION['glpiactiveprofile']['tickettemplates_id'] = $session_tpl_id_back;
         }
      )->contains('src="data:image/png;base64,' . $this->inlineImage . '"');
```